python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -423,6 +438,35 @@ class DbtCliInvocation:
     _stdout: List[Union[str, Dict[str, Any]]] = field(init=False, default_factory=list)
     _error_messages: List[str] = field(init=False, default_factory=list)
 
+    # Caches fetching relation column metadata to avoid redundant queries to the database.
+    _relation_column_metadata_cache: Dict[RelationKey, RelationData] = field(
+        init=False, default_factory=dict
+    )
+
+    def _get_columns_from_dbt_resource_props(
```

Comment:
I think we should separate out the methods for retrieving from the cache and updating the cache.